src/modules/dotnet-publish.ts

```diff
@@ -27,40 +40,75 @@ const DotnetPublish = {
         return `Publishes the dotnet solution from the root of the project (via ${this.cmd()})`;
     },
     getOptions(): OptionStringBuilder {
-        return new OptionStringBuilder("publish", "p");
+        return Options.Publish;
     },
     /**
      * Runs a publish of the dotnet solution to the local file system
      * @param {string} absoluteOutputDir Optional absolute path of release output directory. If not provided,
      * defaults to dotnet solution's 'release' directory
      */
     run(absoluteOutputDir?: string) {
+        DotnetPath.solutionPathOrExit();
+
         if (StringUtils.isEmpty(absoluteOutputDir)) {
             absoluteOutputDir = DotnetPath.releaseDir();
         }
 
-        Echo.message(`Cleaning release directory '${absoluteOutputDir}'...`);
-        shell.rm("-rf", absoluteOutputDir!);
-        Echo.success(" - Successfully cleaned released directory");
-        Echo.newLine();
+        _cleanReleaseDirOrExit(absoluteOutputDir!);
 
         Dir.pushd(DotnetPath.solutionDir()!);
-        Echo.message(
-            `Publishing dotnet solution (via ${this.cmd(absoluteOutputDir)})...`
-        );
 
-        if (shell.exec(this.cmd(absoluteOutputDir).toString()).code !== 0) {
-            Echo.error("Failed to publish dotnet project");
-            shell.exit(1);
-        }
+        _publishSolutionOrExit(absoluteOutputDir!);
 
-        Echo.success(" - Dotnet solution published");
         Dir.popd();
     },
 };
 
 // #endregion Functions
 
+// -----------------------------------------------------------------------------------------
+// #region Private Functions
+// -----------------------------------------------------------------------------------------
+
+const _cleanReleaseDirOrExit = (absoluteOutputDir: string) => {
+    Echo.message(`Cleaning release directory '${absoluteOutputDir}'...`);
+    const { code } = shell.rm("-rf", absoluteOutputDir!);
+
+    if (code === 0) {
+        Echo.success(" - Successfully cleaned released directory");
+        Echo.newLine();
+        return;
+    }
+
+    Echo.error(
+        `Failed to clean release directory '${absoluteOutputDir}': ${code}`
```

Comment:
Is this violating a line length guideline?  If not, I suggest making this a one liner.

Reply:
Yeah, it is - I don't make the rules, Prettier does 😛 I could break it up into a function or variable to prevent the line-breaking, but I've become pretty impartial to the way Prettier decides to break up things at this point.